component/service/src/main/java/org/exoplatform/social/rest/impl/site/SiteRest.java

```diff
@@ -250,6 +257,12 @@ public Response getSiteById(@Context
                               String lang) {
     try {
       PortalConfig site = layoutService.getPortalConfig(Long.parseLong(siteId));
+      if (site == null) {
+        return Response.status(Status.NOT_FOUND).build();
+      }
+      if (!userACL.hasAccessPermission(site)) {
+        return Response.status(Status.UNAUTHORIZED).build();
+      }
```

Comment:
This code will allow anonymous user to find informations about sites : 
By making the search with different siteId, he will be able to find ones corresponding to a real site : 
When search return 404 => it is a not existing id
When search return 401 => it is an existing id.

2 options here : 
- Check the hasAccessPermission before the site == null : it will reduce the possibility but to remove completly the problem
- Return not authorized in both case : it is the solution I prefer

Reply:
`This code will allow anonymous user to find informations about sites`

Nope, this endpoint is protected by `@RolesAllowed("users")`, so not visible publically. Besides, the existence of `siteId` parameter isn't a sensitive information that we will need to protect.

In fact, having a 404 or 401 on URL `/v1/social/sites/1` will not give a sensitive information to the end user. The fact to return the `Not found` and `Not authorized` code is better to give more details about error and make the API more self explanatory.

The fact to return the same response code when resource not found and resource not authorized is effectively relevant when the identifier of the resource itself is security concern and shouldn't be known if it exists or not, like using site name, user names, user emails .... but here for an auto generated and incremented id, no need to avoid the difference in favor of the REST API lisibility.

Reply:
ok, what about the answer about the other concern: the auto incremented id information isn't relevant to protect ? WDYT ?

Reply:
No, it is ok

Reply:
> No, it is ok

Cool, thank. I asked because the `@RolesAllowed` will be removed in Meeds-io/MIPs#100, so just to be sure that it will not be a concern for you when it will be available.